src/preprocessor.ts

```diff
@@ -1,21 +1,22 @@
 import * as crypto from 'crypto';
-import { JestConfig, Path, TransformOptions } from './jest-types';
+import {
+  BabelTransformOptions,
+  CodeSourceMapPair,
+  JestConfig,
+  Path,
+  TransformOptions,
+} from './jest-types';
 import { flushLogs, logOnce } from './logger';
 import { postProcessCode } from './postprocess';
-import {
-  getTSConfig,
-  getTSJestConfig,
-  runTsDiagnostics,
-  injectSourcemapHook,
-} from './utils';
+import { getTSConfig, getTSJestConfig, runTsDiagnostics } from './utils';
 import { transpileTypescript } from './transpiler';
 
 export function process(
   src: string,
   filePath: Path,
   jestConfig: JestConfig,
   transformOptions: TransformOptions = { instrument: false },
-): string {
+): CodeSourceMapPair | string {
```

Comment:
I think this function no longer returns result as a string

Reply:
It does if it's a file we're not processing, see the return src a little further down. It might have been possible just to return an empty sourcemap to avoid the union type, but I didn't want to risk it.